EricGray/homework.js

```diff
@@ -0,0 +1,386 @@
+
+	expectValue(genericArray.join(), "a,b,c", "genericArray.join()");
+
+// Question 3 Object Comparison
+
+//a
+
+var tank = {
+	color: "green",
+	wheels: "tread",
+};
+
+var copy = function(obj) {
+	twin = "";
+	twin = obj;
+	return twin;
+};
+
+var equal = function(objA, objB) {
+	if (objA === objB) {
+		return true
+	} else {
+		return false
+	};
+	};
+
+}
+
+//b
+// try a (for in ) loop
+
+var one = {a:1, b:2, c:3, d:4}
+var two = {a:1, b:3, c:2, d:4}
+
+
+var union = function(objA, objB) {
+	var keysA = Object.keys(objA); //get keys from object and make length possible
+	var lengthObjA = keysA.length; // length for "for" conditional
+	
+
+	var keysB = Object.keys(objB);
+	var lengthObjB = keysB.length;
+
+	var	together = {}; // target object for union
+
+	for (i = 0; i <= lengthObjA - 1; i++)
+		{var valueA = [keysA[i]]; // make valueA touchable
+		var valueB = [keysB[i]];
+		console.log(objA[valueA]);
+		console.log("==" + valueA);
+		console.log("==" + valueB)
+		console.log(objB[valueB]);
+		if (objA[valueA] == objB[valueB]){
+			together[keysA[i]] = objA[valueA]
+		} else {
+			console.log("oh no");
+		};
+		};
+		return together;
+	};
+
+// b try 2 with for (  in  )
+
+//Notes: I want to compare keys directly together, currently (especially notable in
+//the subtraction method, the key 'e' will still compare to key 'f'). Another
+//issue: union doesn't currently add obj 'two's 'f' key:value pair.
+
+var one = {a:1, b:2, c:3, d:4, e: 5}
+var two = {a:1, b:3, c:2, d:4, f: 6}
+var together = {}
+var similar = {}
+var difference = {}
+
+var union = function(objA, objB) {
+	together = {};
+	for (key in objA || key in objB)
+		together[(together[key] = objA[key]) || (together[key] = objB[key])];
+	return together;
+}
+
+var intersection = function(objA, objB) {
+    similar = {};
+    for (key in objA)
+        if (objA[key] == objB[key])
+            similar[key] = objA[key];
+            return similar;
+};
+
+var subtract = function(objA, objB) {
+	difference = {};
+	for (key in objA)
+		difference[key] = objA[key]-objB[key];
+		return difference;
+
+}
+
+//c Sample Tests
+
+
+function assert(claim,message) {
+    if (!claim) console.error(message);
+}
+function expectValue(result, expected, attemptStr) {
+	if (result !== expected) {
+		console.log(attemptStr+" expected result "+expected+", got "+result);
+	}
+}
+function expectNaN(result, attemptStr) {
+	if (!Number.isNaN(result)) {
+		console.log(attemptStr+" expected NaN, got "+result);
+	}
+}
+
+function assert(claim, message) {
+	if (!claim) console.error(message);
+}
+
+function expectValue(result, expected, attemptStr) {
+	if (result !== expected) {
+		console.log(attemptStr+" expected result "+expected+", got "+result);
+	}
+}
+
+expectValue(union(one, two), 'Object {a:1, b:2, c:3, d:4, e: 5}', 'union(one, two)');
+expectValue(union("A", "B"), undefined, 'union(one, two)');
+
+
+
+
+
+
+
+//d Explanations
+
+
+//4 Social Network
+
```

Comment:
Let me know if you want to go over the concepts of the social network function sometime.